deploy.yml

```
@@ -0,0 +1,38 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "main"
+  main:
+    name: Build, Validate and Deploy
```

```suggestion
    name: Build, Validate, and Deploy
```

I'm a fan of the [serial comma](https://en.wikipedia.org/wiki/Serial_comma) ;)